### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by keroloussamy**
In this PR, I made the following 
Your models should include some additional methods.

    Users
        A method that returns the 3 most recent posts for a given user.
    Posts
        A method that updates the posts counter for a user.
        A method which returns the 5 most recent comments for a given post.
    Comments
        A method that updates the comments counter for a post.
    Likes
        A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  has_many :likes
+  has_many :comments
+  belongs_to :user, foreign_key: :users_id
+
+  after_save :update_post_user_counter
+
+  def update_post_user_counter
+    user.update(posts_counter: user.posts.counter)
+  end
```
Comment (Emmyn5600) [inline app/models/post.rb]: Let us increment the post added using this way:

```ruby
  def update_post_counter
    user.increment!(:posts_counter)
  end
```

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :user, foreign_key: :users_id
+  belongs_to :post, foreign_key: :posts_id
+
+  after_save :update_post_like_counter
+
+  def update_post_like_counter
+    post.update(likes_counter: post.likes.counter)
+  end
```
Comment (Emmyn5600) [inline app/models/like.rb]: - [ ] The same implementation goes for the comment count method.

```ruby
def update_likes_counter
  post.increment!(:likes_counter)
end
```

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class Comment < ApplicationRecord
+  belongs_to :user, foreign_key: :users_id
+  belongs_to :post, foreign_key: :posts_id
+
+  after_save :update_post_comment_counter
+
+  def update_post_comment_counter
+    post.update(comments_counter: post.comments.counter)
+  end
```
Comment (Emmyn5600) [inline app/models/comment.rb]: - [ ] Refactor the update_post_comment_counter method with the following:

```ruby
def  update_post_comment_counter
  post.increment!(:comments_counter)
end
```

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts
+  has_many :likes
+  has_many :comments
```
Comment (Emmyn5600) [inline app/models/user.rb]: - [ ] I would recommend adding the foreign_keys for the tables in the user setting up an association, check the below.

```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @keroloussamy 

Good job so far! I know you have worked hard on making this milestone however, There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good PR description :heavy_check_mark:


## Required Changes :recycle:

- [ ]  Check the comment under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @keroloussamy 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉 

- Great job implementing the `after_save` logic to your models 😉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**